benchmark/submit.py

```diff
@@ -0,0 +1,129 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+"""
+Submit benchmark results to a server.
+"""
+
+from __future__ import print_function
+
+import json
+import sys
+
+from twisted.internet import endpoints
+from twisted.internet.defer import succeed
+from twisted.internet.task import react
+from twisted.python import usage
+from twisted.web import client, http
+from twisted.web.http_headers import Headers
+from twisted.web.iweb import IBodyProducer
+
+from zope.interface import implementer
+
+
+@implementer(IBodyProducer)
+class StringProducer(object):
+    def __init__(self, body):
+        self.body = body
+        self.length = len(body)
+
+    def startProducing(self, consumer):
+        consumer.write(self.body)
+        return succeed(None)
+
+    def pauseProducing(self):
+        pass
+
+    def stopProducing(self):
+        pass
+
+
+def submit(agent, result):
```

Comment:
If you intend to make this public and use it elsewhere, perhaps remove the `print` from this function (have that handled by main instead, and errback or callback the deferred depending on if it a success).

On the other hand, if you intend to keep this private, request naming it `_submit` instead of `submit`